src/AmalgamatorSession.ts

```diff
@@ -488,6 +498,56 @@ export class AmalgamatorSession extends LoggingDebugSession {
         }
     }
 
+    protected async disassembleRequest(
+        response: DebugProtocol.DisassembleResponse,
+        args: DebugProtocol.DisassembleArguments
+    ): Promise<void> {
+        if (args.memoryReference) {
+            response.body = {
+                instructions: [],
+            };
+            try {
+                this.addressMap.forEach(
+                    (
+                        [instructionPointerReference, startAdress, endAddress],
+                        childDapID
+                    ) => {
+                        if (
+                            args.memoryReference ===
+                                instructionPointerReference ||
+                            args.memoryReference === startAdress ||
+                            args.memoryReference === endAddress
+                        ) {
+                            this.childDapIndex = childDapID;
+                        }
+                    }
+                );
```

Comment:
Doing my suggestion at line 411 makes this a one liner too:

```typescript
this.childDapIndex = this.addressMap.has(args.memoryReference) ? this.addressMap.get(args.memoryReference) : this.childDapIndex;
```

Reply:
I updated it